eth/consensus/clique/clique.py

```diff
@@ -0,0 +1,171 @@
+                             import_block: BlockHeaderAPI=None) -> ExecutionContext:
+
+    # In Clique consensus, the tx fee goes to the signer
+    fee_recipient = None
+    if import_block is not None:
+        try:
+            fee_recipient = get_block_signer(import_block.header)
+        except Exception:
+            pass
+
+    return ExecutionContext(
+        coinbase=header.coinbase,
+        timestamp=header.timestamp,
+        block_number=header.block_number,
+        difficulty=header.difficulty,
+        gas_limit=header.gas_limit,
+        prev_hashes=prev_hashes,
+        chain_id=chain_context.chain_id,
+        fee_recipient=fee_recipient
+    )
+
+
+def configure_header(vm: VirtualMachineAPI, **header_params: HeaderParams) -> BlockHeaderAPI:
+    with vm.get_header().build_changeset(**header_params) as changeset:
+        # We do this because the default `configure_header` tries to compute the difficulty
+        # which we can not do at this point. We take the `difficulty` as provided and rely
+        # on the `validate_seal` call that will happen as the last step when blocks are
+        # imported.
+        header = changeset.commit()
+    return header
+
+
+class CliqueConsensus:
+    """
+    This class is the entry point to operate a chain under the rules of Clique consensus which
+    is defined in EIP-225: https://eips.ethereum.org/EIPS/eip-225
+    """
+
+    logger = logging.getLogger('eth.consensus.clique.CliqueConsensus')
+
+    def __init__(self, base_db: AtomicDatabaseAPI, epoch_length: int = 30000) -> None:
+        if base_db is None:
+            raise ValueError("Can not instantiate without `base_db`")
+        self._epoch_length = epoch_length
+        self._chain_db = ChainDB(base_db)
+        self._header_cache = HeaderCache(self._chain_db)
+        self._snapshot_manager = SnapshotManager(
+            self._chain_db,
+            self._header_cache,
+            self._epoch_length,
+        )
+
+    def _construct_turn_error_message(self,
+                                      expected_difficulty: int,
+                                      header: BlockHeaderAPI,
+                                      signer: Address,
+                                      signers: Sequence[Address]) -> str:
+
+        return (
+            f"Expected difficulty of {header} to be {expected_difficulty} "
+            f"but was {header.difficulty}.\n"
+            f"Header signer: {encode_hex(signer)}. Valid signers: {signers}"
+        )
+
+    @to_tuple
+    def amend_vm_configuration(self, config: VMConfiguration) -> Iterable[VMFork]:
+        """
+        Amend the given ``VMConfiguration`` to operate under the rules of Clique consensus.
+        """
+        for pair in config:
+            block_number, vm = pair
+            vm_class = vm.configure(
+                extra_data_max_bytes=65535,
+                validate_seal=lambda header: self.validate_seal(header),
```

Comment:
can you see if you can eliminate the lambda using `staticmethod(self.validate_seal)`

Reply:
Works :)